docs/user-guide/how_to_tune_volcano_performance.md

```diff
@@ -0,0 +1,571 @@
+// pkg/controllers/job/job_controller.go:318-333
+func (cc *jobcontroller) belongsToThisRoutine(key string, count uint32) bool {
+    val := cc.genHash(key)
+    return val % cc.workers == count
+}
+
+func (cc *jobcontroller) getWorkerQueue(key string) workqueue.TypedRateLimitingInterface[any] {
+	val := cc.genHash(key)
+	queue := cc.queueList[val%cc.workers]
+	return queue
+}
+
+// genHash source code
+func (cc *jobcontroller) genHash(key string) uint32 {
+    hashVal := fnv.New32() // FNV-1a non-cryptographic hash
+    hashVal.Write([]byte(key))
+    return hashVal.Sum32()
+}
+```
+
+FNV (Fowler–Noll–Vo) is a fast, low-collision non-cryptographic hash function that serves as a **consistent hashing** mechanism in Volcano:
+
+1.  **Serializing Single-Job Operations**: It ensures that the same JobKey is always routed to the same worker, preventing race conditions (like version conflicts or duplicate pod creation) that could arise from multiple goroutines modifying the same Job state concurrently.
+2.  **Load Balancing**: It distributes different Jobs evenly across the `workers` queues, improving parallelism.
+
+> ❓ What if "same Job → same goroutine" is not guaranteed?
+> *   Multiple goroutines could enter the state machine for the same Job simultaneously, causing **Status conflicts** (retries due to ResourceVersion mismatch, optimistic lock failures).
+> *   Duplicate creation of Pods / PodGroups, leading to **resource leaks** and **Gang Scheduling failures**.
+> *   Without this mechanism, a global lock or fine-grained optimistic retries would be necessary, which would be less efficient.
+
+##### PodGroup Creation
+
+```go
+// pkg/controllers/job/job_controller_actions.go:190-214
+func (cc *jobcontroller) createOrUpdatePodGroup(job *batch.Job) error {
+    ...
+    pg := &scheduling.PodGroup{ ... }
+    vcClient.SchedulingV1beta1().PodGroups(...).Create(..., pg, ...)
+    ...
+}
+```
+
+A PodGroup is created with a single API call, and each Job has only one PodGroup. The creation itself is relatively simple (it's just a logical unit) and is unlikely to be a major bottleneck:
+* A PodGroup is essentially a CRD object (with a Spec & Metadata of only a few dozen bytes, as defined in `pkg/controllers/job/job_controller_actions.go`). The creation process is just a single write operation from kube-apiserver to etcd.
+* It does not involve scheduling decisions, node communication, or resource calculations. It returns immediately upon success without subsequent long-running processes.
+
+However, it's worth noting that if `MinMember` is set too high or Queue resources are insufficient, the scheduler may still block Job startup in a **later stage** because the PodGroup conditions are not met. This is a scheduling issue, not a CREATE issue. Additionally, if multiple components (like the Controller and Scheduler) update a PodGroup simultaneously, the operation may fail with a `409 Conflict` error due to a `resourceVersion` mismatch, requiring a retry.
```

Comment:
Just want to ensure that does vc-controller and vc-scheduler still have 409 conflicts on podgroup updating? Which of version of volcano do you use? And did you find which part of codes may cause updating conflict?